src/daft-logical-plan/src/logical_plan.rs

```diff
@@ -266,12 +272,14 @@ impl LogicalPlan {
                 Self::Sample(Sample {fraction, with_replacement, seed, ..}) => Self::Sample(Sample::new(input.clone(), *fraction, *with_replacement, *seed)),
                 Self::Concat(_) => panic!("Concat ops should never have only one input, but got one"),
                 Self::Intersect(_) => panic!("Intersect ops should never have only one input, but got one"),
+                Self::Union(_) => panic!("Union ops should never have only one input, but got one"),
```

Comment:
I think in some cases we must accept single union. We just need to add eliminate single union rule. Just like it's done in datafusion.

Reply:
In this PR, there's no way for an end user to create a single union, so this should never happen.